comdirect/cmd/transaction.go

```diff
@@ -22,16 +25,49 @@ var (
 )
 
 func transaction(cmd *cobra.Command, args []string) {
+	const dateLayout = "2006-01-02"
+	since, err := time.Parse(dateLayout, sinceFlag)
+	if err != nil {
+		log.Fatalf("Failed to parse date from command line: %s", err)
+	}
+
 	client := initClient()
-	ctx, cancel := contextWithTimeout()
-	defer cancel()
+	var transactions = &comdirect.AccountTransactions{}
 
-	options := comdirect.EmptyOptions()
-	options.Add(comdirect.PagingCountQueryKey, countFlag)
-	options.Add(comdirect.PagingFirstQueryKey, indexFlag)
-	transactions, err := client.Transactions(ctx, args[0], options)
+	page := 1
+	pageCount, err := strconv.Atoi(countFlag)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("Can't convert string to int: %s", err)
+	}
+	for {
```

Comment:
Since we are doing a request in each iteration, we need to be careful to not hit the rate limit on the API, which is currently 10 requests per second. A simple solution is to add a timeout after 9 iterations. The drawback is that in case we fetch many transactions, it could take quite some time. Any thoughts on this?

Reply:
Good point, I added rate limiting already on the HTTP client now.